alphabase/psm_reader/psm_reader.py

```diff
@@ -266,17 +213,41 @@ def set_modification_mapping(
             self.modification_mapping = copy.deepcopy(modification_mapping)
 
         self._str_mods_to_lists()
+
+        if self._add_unimod_to_mod_mapping:
+            self._add_all_unimod()
+            self._extend_mod_brackets()
+
         self.rev_mod_mapping = self._get_reversed_mod_mapping()
 
     def _init_modification_mapping(self) -> None:
         self.modification_mapping = {}
 
+    def _add_all_unimod(self) -> None:
+        for mod_name, unimod in MOD_TO_UNIMOD_DICT.items():
+            if mod_name in self.modification_mapping:
+                self.modification_mapping[mod_name].append(unimod)
+            else:
+                self.modification_mapping[mod_name] = [unimod]
+
+    def _extend_mod_brackets(self) -> None:
```

Comment:
This code (ab in general, not your PR) feels to me like we should start a ModificationMapping class 😆

Reply:
Agree. It also depends on how heavy we are going to use these reader classes. Before, I just used them to get unified training data.

Reply:
done here https://github.com/MannLabs/alphabase/pull/247